src/test/java/de/mediathekview/mserver/crawler/kika/tasks/KikaApiTopicDeserializerTest.java

```diff
@@ -1,69 +1,34 @@
 package de.mediathekview.mserver.crawler.kika.tasks;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.reflect.TypeToken;
-
 import de.mediathekview.mserver.crawler.kika.json.KikaApiTopicDto;
 import de.mediathekview.mserver.crawler.kika.json.KikaApiTopicPageDeserializer;
-
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-import static org.junit.Assert.assertEquals;
-
 import java.io.IOException;
 import java.lang.reflect.Type;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.Set;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
-
-@RunWith(Parameterized.class)
 public class KikaApiTopicDeserializerTest {
 
-  private String jsonFile = "";
-  private int numberOfExpectedRecords = 0;
-  private boolean hasSubpage = false;
-  private boolean hasError = false;
-  
-
-  public KikaApiTopicDeserializerTest(String jsonFile, int numberOfExpectedRecords, boolean subpage, boolean error) {
-    this.jsonFile = jsonFile;
-    this.numberOfExpectedRecords = numberOfExpectedRecords;
-    this.hasSubpage = subpage;
-    this.hasError = error;
-  }
-
-  @Parameterized.Parameters
-  public static Object[][] data() {
-    return new Object[][] {
-          {
-            "/kika/KikaApiTopic1.json",
-            10,
-            true,
-            false
-          },
-          {
-            "/kika/KikaApiTopic2.json",
-            11,
-            false,
-            false
-          },
-          {
-            "/kika/KikaApiError.json",
-            0,
-            false,
-            true
-          }
-        };
-
-  }
-
-  @Test
-  public void testDeserializeBrand() throws URISyntaxException, IOException {
+  @ParameterizedTest
+  @MethodSource("getTestData")
```

Comment:
getTestData ist etwas wenig aussagekräftig. Welche Test Data, für was?

Reply:
Fixed in [28eb497](https://github.com/mediathekview/MServer/pull/925/commits/28eb4972ab26425a5193c8e5d69105ff374e82f5)